src/h5web/toolbar/HeatmapToolbar.tsx

```diff
@@ -22,7 +23,7 @@ function HeatmapToolbar(): ReactElement {
     toggleAspectRatio,
     showGrid,
     toggleGrid,
-  } = useHeatmapConfig();
+  } = useHeatmapConfig((state) => state, shallow);
```

Comment:
I noticed some useless re-rendenders and this was the cause.

We call setters in effects without checking if they're changing anything to the state, like:

```ts
  useEffect(() => {
    disableErrors(!errors);
  }, [disableErrors, errors]);
```

This is fine, but if we don't pass a selector to `useLineConfig` and `useHeatmapConfig`, this kind of state update always triggers a re-render. By passing a selector and `shallow`, we check if anything has changed before triggering a re-render.

Somehow I thought zustand did this check internally by default, but that's not the case.